examples/powertools-examples-cloudformation/src/main/java/helloworld/App.java

```diff
@@ -0,0 +1,166 @@
+                log.info("Bucket Deleted {}", bucketName);
+                return Response.success(bucketName);
+            } catch (AwsServiceException | SdkClientException e) {
+                log.error(e);
+                return Response.failed(bucketName);
+            }
+        } else {
+            log.info("Bucket already deleted - no action");
+            return Response.success(bucketName);
+        }
+
+    }
+
+    private boolean bucketExists(String bucketName) {
+        try {
+            HeadBucketResponse headBucketResponse = s3Client.headBucket(HeadBucketRequest.builder().bucket(bucketName).build());
+            if (headBucketResponse.sdkHttpResponse().isSuccessful()) {
+                return true;
+            }
+        } catch (NoSuchBucketException e) {
+            log.info("Bucket does not exist");
+            return false;
+        }
+        return false;
+    }
+
+    private void updateLifecycleRules(String physicalResourceId, int retentionDays) {
+        GetBucketLifecycleConfigurationResponse bucketLifecycleConfiguration = s3Client.getBucketLifecycleConfiguration(GetBucketLifecycleConfigurationRequest.builder().bucket(physicalResourceId).build());
+        log.info(bucketLifecycleConfiguration);
+
+        List<LifecycleRule> oldRules = new ArrayList<>(bucketLifecycleConfiguration.rules());
+        Optional<LifecycleRule> oldExpireLifecycleRule = oldRules.stream().filter(lifecycleRule -> lifecycleRule.id().equals(EXPIRE_OBJECTS_ID)).findFirst();
+        // Check if the existing configuration is equals to the new configuration
+        if (oldExpireLifecycleRule.isPresent()) {
+            if (oldExpireLifecycleRule.get().expiration().days() == retentionDays) {
+                log.info("Retention period is already set to {}. No further actions", retentionDays);
+                return;
+            } else {
+                oldRules.remove(oldExpireLifecycleRule.get());
+            }
+        }
+
+        LifecycleRule updatedLifecycleRule = LifecycleRule.builder().id(EXPIRE_OBJECTS_ID).filter(LifecycleRuleFilter.builder().prefix("").build()).status("Enabled").expiration(LifecycleExpiration.builder().days(retentionDays).build()).build();
+
+        List<LifecycleRule> newRules = new ArrayList<>();
+        newRules.add(updatedLifecycleRule);
+        newRules.addAll(oldRules);
+
+        log.info("Updating Lifecycle configuration");
+        s3Client.putBucketLifecycleConfiguration(PutBucketLifecycleConfigurationRequest.builder().bucket(physicalResourceId).lifecycleConfiguration(BucketLifecycleConfiguration.builder().rules(newRules).build()).build());
+    }
+
+
+    private void createBucketWithRetention(String bucketName, int retentionDays) {
+        S3Waiter waiter = s3Client.waiter();
+        CreateBucketRequest createBucketRequest = CreateBucketRequest.builder().bucket(bucketName).build();
+        s3Client.createBucket(createBucketRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If you don’t explicitly set an AWS region, the SDK consults the default region provider chain to try and determine the region to use.  However, the resulting delay might lead to a poor cold-start. If code portability is not an issue, consider explicitly setting the AWS region using the *region()* method.

[Learn more](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/region-selection.html#automatically-determine-the-aws-region-from-the-environment)